codecs/zarrs.zfp/README.md

```diff
@@ -0,0 +1,88 @@
+# ZFP codec
+
+Defines a `array -> bytes` codec that compresses chunks using the [zfp](https://github.com/LLNL/zfp) algorithm.
+
```

Comment:
zfp natively only supports 1, 2, 3 and 4 dimensional arrays.  However, for zarr it might be useful to allow arrays with more than 4 dimensions provided that at most 4 dimensions have a chunk size greater than 1.  That would allow zfp to be combined with sharding_indexed to handle arbitrary numbers of dimensions.  You would just need to define exactly how the zfp dimensions are determined from the zarr dimensions. (See also https://github.com/seung-lab/zfpc which is supported by the neuroglancer-precomputed format --- with zarr I think an extra container format is unnecessary because sharding_indexed already serves that purpose quite well)

Reply:
My suggestion for defining that would be to say: if the number of dimensions with chunk size > 1 is within the range `[1, 4]`, those are the zfp dimensions.    If there are more than 4 such dimensions, it is an error.  If there are zero such dimensions, i.e. the chunk contains just a single element, treat it as a 1-d array of 1 element (so that users don't have to worry about this as a special case).

Reply:
Great catch, and a good idea in regards to supporting higher dimensional arrays. I'll add support for that in `zarrs`.

I wasn't aware of https://github.com/seung-lab/zfpc, but it highlights the reason why I didn't include complex numbers for this codec. A Zarr V3 way to do that could be an array-to-bytes codec that deinterleaves complex data types and encodes them independently.

Reply:
Regarding your "may support" change:  what I described (and what you have) is a breaking change, because previously if you had a chunk size of [64, 64, 1, 64] it would get encoded as a 4-d zfp (effectively padded to [64, 64, 4, 64]) --- I'm not sure exactly what the implications are in terms of runtime performance and compression ratio, but it seems plausible that it may make compression and decompression 4x as slow.  Under the new language, it would instead get encoded as a 3-d zfp with chunk shape `[64, 64, 64]`.  Without the header I don't know that this can be detected automatically.  For zarrs, assuming backwards compatibility is important, perhaps you can support the old behavior under the old identifier that you used, and the new behavior under the plain `zfp` identifier.

Reply:
In that case, always maintaining the chunk shape for <=4D data and only collapsing dimensions for 5D+ data might be a good option. I'll trial this in implementation, and come back to this PR afterwards.